### PR TITLE
Milestone4/logic

### PR DESCRIPTION
**Pull request by YoseptF**
I added the logic to run the calculator
- The operator module uses big.js to run math operations
- The calculate module uses operator to modify the state of the application

I also modified some of the existing logic in the app to make it easier to use these 2 new modules.

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import Big from 'big.js';
+
+const operate = (numberOne, numberTwo, operation) => {
+  const bigOne = Big(numberOne);
+  const bigTwo = Big(numberTwo);
+  switch (operation) {
+    case '-':
+      return bigOne - bigTwo;
```
Comment (bellom) [inline src/logic/operate.js]: Good job getting this far.
The way you used the new `big.js` I am curious about it.
kindly check out what we have on the doc.
`https://mikemcl.github.io/big.js/`
Example:
`bigOne.minus(bigTwo)`

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    case '+':
+      return bigOne + bigTwo;
+    case 'X':
+      return bigOne * bigTwo;
+    case '%':
+      return bigOne % bigTwo;
```
Comment (bellom) [inline src/logic/operate.js]: Note: case `%` require just the first digit which is `bigOne`

File: src/logic/calculate.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      newData.operation = null;
+      break;
+
+    default:
+      newData.total = operate(...dataObject);
+      newData.next = '0';
+      newData.operation = null;
+      break;
+  }
+  return newData;
```
Comment (bellom) [inline src/logic/calculate.js]: we need to add more conditions, like;
1. division zero error handling(if we divide a digit by zero we should get an error message)
2. when `.` return `0.`

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @YoseptF  :clap: 

### Awesome job so far!! :+1: 

**NOTE: kindly fix the linter's error**
fix all errors:** checkout this [linters-config#troubleshooting](https://github.com/microverseinc/linters-config#troubleshooting) for help. Example below;
image
#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by AdnanAfsari — APPROVED** [review]
**Status: Approved**

Hello @YoseptF ✋ 

Nice work, You have made changes requested by the previous reviewer.It's a 👍 .

Happy Coding
Adnan Afsari

**Pull request merged**